tools/ktx/command_compare.cpp

```diff
@@ -0,0 +1,1929 @@
+            } else
+                printIndent(2, "null{}", nl);
+            printIndent(1, "]");
+        }
+    }
+
+    void operator<<(const DiffTextCustom& diff) {
+        if (!diff.isDifferent()) return;
+        different = true;
+
+        assert(outputFormat == OutputFormat::text);
+        if (diff.hasText(0))
+            printIndent(0, "-{}\n", diff.text(0));
+        if (diff.hasText(1))
+            printIndent(0, "+{}\n", diff.text(1));
+    }
+
+    void operator<<(const DiffMismatch& diff) {
+        different = true;
+
+        const auto space = outputFormat != OutputFormat::json_mini ?  " " : "";
+        const auto nl = outputFormat != OutputFormat::json_mini ?  "\n" : "";
+
+        if (outputFormat == OutputFormat::text) {
+            printIndent(0, "+{}\n", diff.textMsg);
+        } else {
+            printIndent(0, ",{}", nl);
+            printIndent(1, "\"{}\":{}[]", diff.jsonPath, space);
+        }
+    }
+};
+
+// -------------------------------------------------------------------------------------------------
+
+/** @page ktx_compare ktx compare
+@~English
+
+Compare two KTX2 files.
+
+@section ktx_compare_synopsis SYNOPSIS
+    ktx compare [option...] @e input-file1 @e input-file2
+
+@section ktx_compare_description DESCRIPTION
+    @b ktx @compare compares the two KTX2 files specified as the @e input-file1 and @e input-file2
+    arguments and outputs any mismatch in texture information and/or image data.
+    The command implicitly calls @ref ktx_validate "validate" and prints any found errors
+    and warnings to stdout.
+    If any of the specified input files are invalid then comparison is done based on best effort
+    and may be incomplete.
+
+    The JSON output formats conform to the https://schema.khronos.org/ktx/compare_v0.json
+    schema even if the input file is invalid and certain information cannot be parsed or
+    displayed.
+    Additionally, for JSON outputs the KTX file identifier is printed using "\u001A" instead of
+    "\x1A" as an unescaped "\x1A" sequence inside a JSON string breaks nearly every JSON tool.
+    Note that this does not change the value of the string only its representation.
+
+    @note @b ktx @b compare prints using UTF-8 encoding. If your console is not
+    set for UTF-8 you will see incorrect characters in output of the file
+    identifier on each side of the "KTX nn".
+
+    The following options are available:
+    @snippet{doc} ktx/command.h command options_format
+    <dl>
+        <dt>\--content raw | image | ignore</dt>
+        <dd>Controls how image content is compared. Possible options are: <br />
+            @b raw - Encoded image data is compared verbatim, as it appears in the file. <br />
+            @b image - Effective image data is compared per texel block. <br />
+            @b ignore - Ignore image contents. <br />
+            The default mode is @b raw, meaning that the encoded image data must match exactly.
```

Comment:
Should explain "effective image data" in an additional sentence following "exactly."

Reply:
This part will go through some major changes in phase 2, so I'll leave it as is for now, because a lot of things will get clarified in the meantime.